libs/core/core.cpp

```diff
@@ -467,15 +467,38 @@ TNumber atan2(TNumber y, TNumber x) {
 }
 
 //%
-int random(int max) {
-    if (max == INT_MIN)
-        return -device.random(INT_MAX);
-    else if (max < 0)
-        return -device.random(-max);
-    else if (max == 0)
-        return 0;
-    else
-        return device.random(max);
+TNumber random() {
+    return fromDouble(device.random(INT_MAX) / (double)INT_MAX);
+}
+
+//%
+TNumber randomRange(TNumber min, TNumber max) {
+    if (isNumber(min) && isNumber(max)) {
+        int mini = numValue(min);
+        int maxi = numValue(max);
+        if (mini > maxi) {
+            int temp = mini;
+            mini = maxi;
+            maxi = temp;
+        }
+        if (maxi == mini)
+            return fromInt(mini);
+        else
+            return fromInt(mini + device.random(maxi - mini + 1));
+    }
+    else {
+        double mind = toDouble(min);
+        double maxd = toDouble(max);
+        if (mind > maxd) {
+            double temp = mind;
+            mind = maxd;
+            maxd = temp;
+        }
+        if (maxd == mind)
+            return fromDouble(mind);
+        else
+            return fromDouble(mind + device.random(maxd - mind));
```

Comment:
oh, for example for randomRange(0.5, 3.5) this will return 0.5, 1.5, 2.5 or 3.5, but never say 1.83434.